### ISSUE TITLE
C-c C-o goes directly to match instead of occur buffer in ivy-imenu-anywhere

### ISSUE THREAD

**Issue by unhammer** [vspinu/imenu-anywhere#40]
(Originally posted at https://github.com/abo-abo/swiper/issues/1306 )

If I do `M-x ivy-imenu-anywhere` and hit `C-c C-o`, it will go to the match (as if I'd hit enter), instead of opening the occur buffer (the occur buffer is still opened, but "behind" the buffer with the match).

**Comment by vspinu** [comment]
I don't think this is specific to `imenu-anywhere`.  It [does a tiny bit more](https://github.com/vspinu/imenu-anywhere/blob/master/imenu-anywhere.el#L308) than replacing `completing-read-function` for the duration of the call.  I hope @abo-abo  could tell more about why `ivy-occur` (C-c C-o) fails in that situation.

**Comment by abo-abo** [comment]
OK, I see the issue now. Using just `completing-read-function` won't work with `ivy-occur` - `ivy-read` needs to be called directly (to pass the `action` parameter).

**Comment by vspinu** [comment]
Thanks Oleh. So, is there anything I can do on `imenu-anywhere` side at the moment?

**Comment by abo-abo** [comment]
If it's possible, try to use `ivy-read` instead of `completing-read-function` (when `ivy-mode` is on).
See `counsel-imenu`: it works with `ivy-occur` because it passes the `:action` parameter to `ivy-read`.
